File: SafetyProgram.DocumentObjectUi/ChemicalTableNs/View/ChemicalTableViewModel.cs
using System;
using System.Collections.Generic;
using SafetyProgram.Base.Interfaces;
using SafetyProgram.Models;

namespace SafetyProgram.DocumentObjectUi.ChemicalTableNs.View
{
    public sealed class ChemicalTableViewModel : IChemicalTableViewModel
    {
        public ChemicalTableViewModel(IChemicalTable chemicalTable,
            IConfiguration configuration,
            ICommandInvoker commandInvoker)
        {
            //monitor the kernel etc.
        }

        public string Header
        {
            get
            {
                throw new NotImplementedException();
            }
            set
            {
                throw new NotImplementedException();
            }
        }

        public System.Windows.Controls.ContextMenu ContextMenu
        {
            get { throw new NotImplementedException(); }
        }

        public System.Collections.ObjectModel.ObservableCollection<Models.ICoshhChemical> Chemicals
        {
            get { throw new NotImplementedException(); }
        }

        public System.Collections.ObjectModel.ObservableCollection<Models.ICoshhChemical> SelectedChemicals
        {
            get { throw new NotImplementedException(); }
        }

        public List<System.Windows.Input.InputBinding> Hotkeys
        {
            get { throw new NotImplementedException(); }
        }

        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
    }
}

File: SafetyProgram.UI.ViewModels/ChemicalTableViewModel.fs
namespace SafetyProgram.UI.ViewModels

open SafetyProgram.UI.Models
open System.Collections.ObjectModel

type ChemicalTableViewModel(model : GuiChemicalTable) = 

    let propertyChangedEvent = new Event<_,_>()
    let selectedChemicals = new ObservableCollection<_>()

    interface IChemicalTableViewModel with
        member this.Header 
            with get () = model.Header
            and set x = model.Header <- x

        member this.Chemicals
            with get () = model.Chemicals   
            
        member this.SelectedChemicals
            with get () = selectedChemicals

        [<CLIEvent>]
        member this.PropertyChanged = propertyChangedEvent.Publish

        member this.Dispose() = ()

    // IMPLICIT IMPL
    member this.Header 
            with get () = model.Header
            and set x =
                model.Header <- x

    member this.Chemicals
        with get () = model.Chemicals            

    [<CLIEvent>]
    member this.PropertyChanged = propertyChangedEvent.Publish

    member this.Dispose() = ()